model/src/cfd.rs

```diff
@@ -1022,18 +1023,66 @@ impl Cfd {
         Ok(settlement)
     }
 
-    pub fn propose_collaborative_settlement(
-        &self,
+    pub fn start_close_position_taker(
+        self,
         current_price: Price,
         n_payouts: usize,
-    ) -> Result<(CfdEvent, SettlementProposal)> {
+    ) -> Result<(CfdEvent, ClosePositionTransaction, SettlementProposal)> {
         anyhow::ensure!(
             !self.is_in_collaborative_settlement()
                 && self.role == Role::Taker
                 && self.can_settle_collaboratively(),
             "Failed to propose collaborative settlement"
         );
 
+        let (close_position_tx, proposal) = self.make_proposal(current_price, n_payouts)?;
+
+        Ok((
+            CfdEvent::new(
+                proposal.order_id,
+                EventKind::CollaborativeSettlementStarted { proposal },
+            ),
+            close_position_tx,
+            proposal,
+        ))
+    }
+
+    /// Use this function after receiving settlement proposal
+    pub fn start_close_position_maker(
+        self,
+        current_price: Price,
+        n_payouts: usize,
+        proposed_close_transaction: Transaction,
+    ) -> Result<(CfdEvent, ClosePositionTransaction, SettlementProposal)> {
+        anyhow::ensure!(
+            !self.is_in_collaborative_settlement()
+                && self.role == Role::Maker
+                && self.can_settle_collaboratively(),
+            "Failed to start collaborative settlement"
+        );
+
+        let (close_position_tx, proposal) = self.make_proposal(current_price, n_payouts)?;
+
+        anyhow::ensure!(
+            close_position_tx.unsigned_transaction == proposed_close_transaction,
+            "Proposed close transaction does not equal locally created one"
+        );
```

Comment:
:upside_down_face: It would be good to see the difference between the two transactions here.